test/Microsoft.AspNetCore.Routing.Tests/TestObjects/TestMatcher.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing.Matchers;
+
+namespace Microsoft.AspNetCore.Routing.Tests.TestObjects
+{
+    internal class TestMatcher : Matcher
+    {
+        private readonly bool _isHandled;
+
+        public TestMatcher(bool isHandled)
+        {
+            _isHandled = isHandled;
+        }
+
+        public override Task MatchAsync(HttpContext httpContext, IEndpointFeature feature)
+        {
+            if (_isHandled)
+            {
+                feature.Endpoint = new TestEndpoint(EndpointMetadataCollection.Empty, "Test endpoint");
```

Comment:
Is this temporary? because endpoints are currently exposed through DispatcherOptions, so i think the test setup should reflect that.

Reply:
Do you think it adds value doing that in a testing mock though?

Reply:
Flowing an endpoint from the DispatcherOptions would be a test for the real matcher. This is a minimum viable product matcher.

But it can evolve in future as we need it to.